examples/feature_demo/compute_cloth.py

```diff
@@ -0,0 +1,703 @@
+        data=forces, usage=wgpu.BufferUsage.STORAGE | wgpu.BufferUsage.COPY_DST
+    )
+
+    # Verlet vertex params buffer (is_fixed, spring_count, spring_pointer)
+    spring_list = []
+    vertex_params = np.zeros((vertex_count, 3), dtype=np.uint32)
+
+    for i, vertex in enumerate(verlet_vertices):
+        vertex_params[i, 0] = 1 if vertex["is_fixed"] else 0
+        vertex_params[i, 1] = len(vertex["spring_ids"])
+        vertex_params[i, 2] = len(spring_list)
+        spring_list.extend(vertex["spring_ids"])
+
+    vertex_params_buffer = gfx.Buffer(
+        data=vertex_params, usage=wgpu.BufferUsage.STORAGE | wgpu.BufferUsage.COPY_DST
+    )
+
+    # spring list buffer
+    spring_list_buffer = gfx.Buffer(
+        data=np.array(spring_list, dtype=np.uint32),
+        usage=wgpu.BufferUsage.STORAGE | wgpu.BufferUsage.COPY_DST,
+    )
+
+    # spring data buffers
+    spring_vertex_ids = np.array(
+        [[s["vertex0_id"], s["vertex1_id"]] for s in verlet_springs], dtype=np.uint32
+    )
+    spring_rest_lengths = np.array(
+        [s["rest_length"] for s in verlet_springs], dtype=np.float32
+    )
+
+    spring_vertex_ids_buffer = gfx.Buffer(
+        data=spring_vertex_ids,
+        usage=wgpu.BufferUsage.STORAGE | wgpu.BufferUsage.COPY_DST,
+    )
+
+    spring_rest_lengths_buffer = gfx.Buffer(
+        data=spring_rest_lengths,
+        usage=wgpu.BufferUsage.STORAGE | wgpu.BufferUsage.COPY_DST,
+    )
+
+    # spring forces buffer
+    spring_forces = np.zeros((spring_count, 3), dtype=np.float32)
+    spring_force_buffer = gfx.Buffer(
+        data=spring_forces, usage=wgpu.BufferUsage.STORAGE | wgpu.BufferUsage.COPY_DST
+    )
+
+    # param buffer [dampening, sphere_pos(3), sphere_active, wind, stiffness, time]
+    params_data = np.array([0.99, 0.0, 0.0, 0.0, 1.0, 1.0, 0.2, 0.0], dtype=np.float32)
+    params_buffer = gfx.Buffer(
+        data=params_data, usage=wgpu.BufferUsage.UNIFORM | wgpu.BufferUsage.COPY_DST
+    )
+
+    return (
+        position_buffer,
+        force_buffer,
+        vertex_params_buffer,
+        spring_list_buffer,
+        spring_vertex_ids_buffer,
+        spring_rest_lengths_buffer,
+        spring_force_buffer,
+        params_buffer,
+        vertex_count,
+        spring_count,
+    )
+
+
+# setup Verlet geometry and buffers
+setup_verlet_geometry()
+
+(
+    position_buffer,
+    force_buffer,
+    vertex_params_buffer,
+    spring_list_buffer,
+    spring_vertex_ids_buffer,
+    spring_rest_lengths_buffer,
+    spring_force_buffer,
+    params_buffer,
+    vertex_count,
+    spring_count,
+) = setup_verlet_buffers()
+
+# compute shaders for Verlet integration
+spring_compute_wgsl = """
+struct Params {
+    dampening: f32,
+    sphere_x: f32,
+    sphere_y: f32,
+    sphere_z: f32,
+    sphere_active: f32,
+    wind: f32,
+    stiffness: f32,
+    time: f32,
+}
+
+@group(0) @binding(0) var<uniform> params: Params;
+@group(0) @binding(1) var<storage, read> positions: array<f32>;
```

Comment:
Here,  the positions buffer should actually be of type array<vec3<f32>>, However, due to data alignment reasons, we do not support it, so we can only flatten it. 
As you can see in the code below, it is not very convenient for both data reading and storage.